src/serialize.jl

```diff
@@ -0,0 +1,27 @@
+using ..ChemistryFeaturization.AbstractType: AbstractAtoms, AbstractFeaturization
+
+"""
+    SerializableEncodedFeature
+
+Helps in serialization, and keeps a record of the `featurization` applied to an `atoms`, and the `encoded_features`
+that are generated as a result.
+
+Note: `encoded_features` will NOT change for a given atoms-featurization pair.
+"""
+struct SerializableEncodedFeature
+    atoms::AbstractAtoms
+    featurization::AbstractFeaturization
+    encoded_features::Any
+
+    # inner constructor - generates `encoded_features`
+    SerializableEncodedFeature(atoms::AbstractAtoms, featurization::AbstractFeaturization) = new(atoms, featurization, featurize(atoms, featurization))
+end
+
+function decode(sef::SerializableEncodedFeature)
+    decoded = decode(sef.featurization, sef.encoded_features)
+    # is this loop universally applicable?
```

Comment:
perhaps worth making the new container (currently SEF) a parametric type so that we can dispatch on the types of `Atoms` and featurizations in case it sometimes isn't?

Reply:
> perhaps worth making the new container (currently SEF) a parametric type so that we can dispatch on the types of Atoms and featurizations in case it sometimes isn't?

I'm not sure I follow. In case it sometimes isn't what?

Reply:
haha sorry, I meant having a parametric type like `SerializableEncodedFeature{AtomsType, FeaturizationType}` and then we can dispatch the decode function on those types if needed in case that loop ends up being not universally applicable.

Reply:
Resolved in fa8595b.